source/object-retention/minio-object-locking.rst

```diff
@@ -0,0 +1,446 @@
+.. _minio-object-locking:
+
+====================
+MinIO Object Locking
+====================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+Overview
+--------
+
+MinIO Object Locking ("Object Retention") enforces Write-Once Read-Many (WORM)
+immutability to protect :ref:`versioned objects <minio-bucket-versioning>` from
+deletion. MinIO supports both 
+:ref:`duration based object retention <minio-object-locking-retention-modes>` 
+and 
+:ref:`indefinite Legal Hold retention <minio-object-locking-legalhold>`.
+
+MinIO Object Locking provides key data retention compliance and meets
+SEC17a-4(f), FINRA 4511(C), and CFTC 1.31(c)-(d) requirements as per 
+`Cohasset Associates <https://min.io/cohasset?ref-docs>`__.
+
+.. card-carousel:: 1
+
+   .. card:: Bucket Without Locking
+
+      .. image:: /images/retention/minio-versioning-delete-object.svg
+         :alt: Deleting an Object
+         :align: center
+
+      MinIO versioning preserves the full history of object mutations. 
+      However, applications can explicitly delete specific object versions.
+
+   .. card:: Bucket With Locking
+
+      .. image:: /images/retention/minio-object-locking.svg
+         :alt: 30 Day Locked Objects
+         :align: center
+
+      Applying a default 30 Day WORM lock to objects in the bucket ensures
+      a minimum period of retention and protection for all object versions.
+
+   .. card:: Delete Operations in Locked Bucket
+
+      .. image:: /images/retention/minio-object-locking-delete.svg
+         :alt: Delete Operation in Locked Bucket
+         :align: center
+
+      Delete operations follow normal behavior in 
+      :ref:`versioned buckets <minio-bucket-versioning-delete>`, where MinIO
+      creates a ``DeleteMarker`` for the object. However, non-Delete Marker 
+      versions of the object remain under the retention rules and are protected 
+      from any specific deletion or overwrite attempts.
+
+   .. card:: Versioned Delete Operations in Locked Bucket
+
+      .. image:: /images/retention/minio-object-locking-delete-version.svg
+         :alt: Versioned Delete Operation in a Locked Bucket
+         :align: center
+
+      MinIO blocks any attempt to delete a specific object version held under
+      WORM lock. The earliest possible time after which a client may delete
+      the version is when the lock expires.
+
+MinIO object locking is 
+:s3-docs:`feature and API compatible with AWS S3 <object-lock.html>`. 
+This page summarizes Object Locking / Retention concepts as implemented by 
+MinIO. See the AWS S3 documentation on
+:s3-docs:`How S3 Object Lock works <object-lock.html>` for additional
+resources.
+
+You can only enable object locking during bucket creation as per 
```

Comment:
@eco-minio LMK what you think, this should make the behavior as clear as possible.

Reply:
looks good